prime-dotnet-webapi/Services/SiteService.cs

```diff
@@ -435,9 +435,15 @@ public async Task<BusinessLicence> UpdateBusinessLicenceAsync(int siteId, Busine
             return businessLicence;
         }
 
-        public async Task<BusinessLicenceDocument> AddBusinessLicenceDocumentAsync(int siteId, Guid documentGuid)
+        public async Task<BusinessLicenceDocument> AddOrReplaceBusinessLicenceDocumentAsync(int businessLicenceId, Guid documentGuid)
         {
-            var businessLicence = await _context.BusinessLicences.Where(bl => bl.SiteId == siteId).SingleOrDefaultAsync();
+            var businessLicence = await _context.BusinessLicences
+                .Include(bl => bl.BusinessLicenceDocument)
+                .Where(bl => bl.Id == businessLicenceId).SingleOrDefaultAsync();
```

Comment:
You can simplify this line to just `.SingleOrDefaultAsync(bl => bl.Id == businessLicenceId);`